### DISCUSSION TITLE
resteasy-reactive: is it possible to stream output directly?

### DISCUSSION THREAD

**Discussion by wgresshoff** [quarkusio/quarkus#26473] [Q&A] [answered]
Hello,
with resteasy Classic it was possible to directly stream contents like this:
`        StreamingOutput stream = output -> {
            try {
                builder.toStream(output)
                        .run();
            } catch (Exception e) {
                logger.error(e.getMessage(), e);
            }
        };
        return Response.ok(stream, MediaType.APPLICATION_OCTET_STREAM)
                .header("Content-Disposition",
                        "attachment; filename=\"" + fileName + "\"" )
                .build();
`
builder is a PDFRenderer that converts HTML to PDF. 
However, with resteasy Reactive I don't see how to achieve this without first generating a temporary file, as StreamingOutput is not supported. Is it still possible? And if, how would I do it?
Currently I'm using Quarkus 2.8.1.

Best regards
Werner

[a comment by an automated account is not shown]

**Comment by geoand** [comment, marked as answer]
`StreamingOutput` is **not** supported. See https://quarkus.io/guides/resteasy-reactive#streaming-support for your options

**Reply by wgresshoff** [reply]
That's exactly the guide where I didn't find the answer to my question. Or perhaps I just didn't understand it.

**Reply by FroMage** [reply]
Well, it's more complicated than that, because you have an API which is streaming while blocking, meaning that even if you find a way to stream the data using a reactive type such as `Multi` your producer (`builder`) needs to run the whole time and cannot be interrupted and resumed when the client is ready for IO. So you could run this on a blocking endpoint, but it will lock up a thread for the entire duration of the streaming. Then it depends whether you can generate the PDF faster than you can send it to your client. If yes, then probably better to produce it to memory or a temp file, and stream that. It really depends on your use-case, but there's no one-size-fits-all.

**Reply by wgresshoff** [reply]
Thank you for your great explanation! For the time being, I generate a temp file. I now have a far better understanding of non-blocking/blocking API's.
